frontend/src/slices/gameDataSlice.ts

```diff
@@ -17,11 +18,14 @@ export const gameDataSlice = createSlice({
   reducers: {
     addBit: (state) => {
```

Comment:
The `addBits` function actually does the same thing as the `addBit` function with extra functionality. You could refactor this so that every dispatch of `addBit` actually calls `addBits` but just passes in `1` for the `additionalBits` property.